### ISSUE TITLE
Investigate use of Keras core

### ISSUE THREAD

**Issue by ml-evs** [ppdebreuck/modnet#158]
Keras Core is now in beta https://keras.io/keras_core/, and will replace the current `tf.keras`API in v3. This would allow us to make pytorch, JAX and TF models with the same API, and may lead to performance improvements. Probably this will also break lots of backwards-compat of model loading (see #155...) and might need a few more code changes for custom loss functions etc. I will try to find time to investigate this before MODNet v1.0.

**Comment by ml-evs** [comment]
This is now officially released and I think I'd like to try implementing it. Looks like it could cover lots of cases for us (e.g., model parallelism)
